src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -0,0 +1,134 @@
+                    self.change(req, 'staged', '+++ monit')
+                # if pileup transfers are completed AND
+                # some input blocks are completed,
+                # move the request status staging -> staged
+                elif self.pileupTransfersCompleted(transferRecords):
+                    self.logger.debug(
+                        "+++ request %s pileup transfers are completed", req)
+                    self.change(req, 'staged', '+++ monit')
+                # transfers not completed
+                # just update the database with their completion
+                else:
+                    self.logger.debug(
+                        "+++ request %s transfers are %s completed",
+                        req, completion)
+                    # for us workflow name is the same as request name
+                    wname = reqName
+                    requestStatuses[wname] = transferRecords
+            self.updateTransferInfo(requestStatuses)
+        except Exception as err:  # general error
+            self.logger.exception('+++ monit error: %s', str(err))
+
+    def getTransferRecords(self, reqName):
+        """
+        Get transfer records for given request name from CouchDB.
+        Transfer records on backend has the following form
+        https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
+        So far logic is based on option D of the records
+
+        .. doctest::
+
+            {"workflowName": "bla",
+             "timestamp": 123,
+             "transfers": [rec1, rec2, ... ]
+            }
+            where each record has the following format:
+            {"dataset":"/a/b/c", "dataType": "primary", "transferIDs": [1,2], "completion":0}
+
+        :param reqName: name of the request
+        :return: list of of status records
+        """
+        transferRecords = []
+
+        # in our case workflow name is the same as request name
+        wname = reqName
+        # get existing transfer IDs record
+        doc = self.getTransferInfo(wname)
+        if not doc:
+            self.logger.error("Failed to find a transfer document for request: %s", wname)
+            return transferRecords
+
+        # loop over all records and obtain new transfer IDs
+        for rec in doc['transfers']:
+            dataset = rec['dataset']
+            dtype = rec['dataType']
+            # obtain new transfer ids for given request name and dataset
+            tids, completion = self.getTransferIds(dataset)
+            rec = {'dataset': dataset, 'dataType': dtype,
+                   'transferIDs': tids, "completion": completion}
+            transferRecords.append(rec)
+        return transferRecords
+
+    def pileupTransfersCompleted(self, transferRecords):
+        "Check if pileup transfers are completed for given transfer records"
+        # loop over all records and obtain new transfer IDs
+        thr = self.msConfig.get('completedThreshold', 100)
```

Comment:
same comment here, please hard-wire it to 100. It's anyways wrong, it should use unified config, not the microservice one.

Reply:
done